tests/basic/mutex.rs

```diff
@@ -6,18 +6,21 @@ use test_env_log::test;
 
 #[test]
 fn basic_lock_test() {
```

Comment:
is this change intentional?

Reply:
Do you mean why did I change from `check` to `check_dfs` ?  I think I was trying out something in an early revision; can revert that back.

Reply:
Yeah. We should make this change, but in a separate PR, because we can remove the round-robin scheduler at the same time.

Reply:
Reverted back to the old version.